api/client/auth0/store.go

```diff
@@ -0,0 +1,107 @@
+package auth0
+
+import (
+	"errors"
+	"io/fs"
+
+	zedfs "github.com/brimdata/zed/pkg/fs"
+	"github.com/brimdata/zed/pkg/storage"
+)
+
+func normalizeLakeURI(lake string) string {
+	u := storage.MustParseURI(lake)
+	return u.String()
+}
+
+type Store struct {
+	path string
+}
+
+func NewStore(path string) *Store {
+	return &Store{path}
+}
+
+func (s Store) LakeTokens(lake string) (*Tokens, error) {
+	creds, err := s.load()
+	if err != nil || creds == nil {
+		return nil, err
+	}
+	return creds.ServiceTokens(lake), nil
+}
+
+func (s Store) SetLakeTokens(lake string, tokens Tokens) error {
+	creds, err := s.load()
+	if err != nil {
+		return err
+	}
+	creds.AddTokens(lake, tokens)
+	return s.save(creds)
+}
+
+func (s Store) RemoveLakeTokens(lake string) error {
+	creds, err := s.load()
+	if err != nil {
+		return err
+	}
+	return s.save(creds)
+}
+
+func (s Store) load() (*Credentials, error) {
+	var creds Credentials
+	if err := zedfs.UnmarshalJSONFile(s.path, &creds); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return &Credentials{}, nil
+		}
+		return nil, err
+	}
+	return &creds, nil
+}
+
+func (s Store) save(creds *Credentials) error {
+	return zedfs.MarshalJSONFile(creds, s.path, 0600)
+}
+
+type ServiceInfo struct {
+	URL    string `json:"url"`
+	Tokens Tokens `json:"tokens"`
+}
+
+type Credentials struct {
+	Version  int           `json:"version"`
+	Services []ServiceInfo `json:"services"`
```

Comment:
Any reason not to simplify by changing this to a URL-keyed map?

Reply:
Avoiding a migration here, but yes we should do this. I'll do this in a follow up pr that will make the auth0 endpoint to services endpoints.

Reply:
> I'll do this in a follow up pr that will make the auth0 endpoint to services endpoints.
I don't follow. What did you mean here?

Reply:
We need to really map the auth0 api endpoint to each token instead of the service uri. As is (with my refresh token pr) if you have an auth0 authenticated lake service and you authenticate a different lake service using the same auth0 api endpoint, the previous access token is invalidated. Actually I might as well address this in my refresh token pr.